antsibull/schemas/plugin.py

```diff
@@ -12,28 +12,112 @@
 
 import pydantic as p
 
-from .base import (OPTION_TYPE_F, REQUIRED_ENV_VAR_F, RETURN_TYPE_F, BaseModel, DeprecationSchema,
-                   DocSchema, JSONValueT, LocalConfig, OptionsSchema, list_from_scalars,
+from .base import (OPTION_TYPE_F, REQUIRED_ENV_VAR_F, RETURN_TYPE_F, VERSION_F, DATE_F,
+                   COLLECTION_NAME_F, REQUIRED_COLLECTION_NAME_F, BaseModel, DocSchema,
+                   JSONValueT, LocalConfig, OptionsSchema, list_from_scalars,
                    normalize_option_type_names, transform_return_docs)
 
 _SENTINEL = object()
 
 
+class OptionDeprecationSchema(BaseModel):
+    """Schema for Option Deprecation Fields."""
+
+    version: str = VERSION_F
+    date: str = DATE_F
+    collection_name: str = REQUIRED_COLLECTION_NAME_F
+    why: str
+    alternative: str = ''
+
+    @p.root_validator(pre=True)
+    def rename_version(cls, values):
+        """Make deprecations at this level match the toplevel name."""
+        removed_in = values.get('removed_in', _SENTINEL)
+        if removed_in is not _SENTINEL:
+            if values.get('version'):
+                raise ValueError('Cannot specify `removed_in` if `version`'
+                                 ' has been specified.')
+
+            values['version'] = removed_in
+            del values['removed_in']
+
+        return values
```

Comment:
From the docstring, it sounds like we want everything to match the toplevel name for this, which is `removed_in`.

So right now, the way I've written these Schemas, the produced json schema should allow the name that we wish it to be.  Current practice would be invalid in the json schema.  The `root_validator` takes care of transforming from the current practice into our desired format.  (if this gets turned into a test, a lot of things will have to start off in the ignore list and get ported over time.  If this gets turned into schema validation code in ansible-doc, it would need to be a normalizing schema just like this which converts from the old names to a standardized name).

So, for instance, the portion of this Schema that deals with version should look like:
```
class OptionDeprecationSchema(BaseModel):
    removed_in: str = VERSION_F

    @p.root_validator(pre=True)
    def rename_version(cls, values):
        """Make deprecations at this level match the toplevel name."""
        version = values.get('version', _SENTINEL)
        if version is not _SENTINEL:
            if values.get('removed_in', _SENTINEL) is not _SENTINEL:
                raise ValueError('Cannot specify `removed_in` if `version`'
                                 ' has been specified.')

            values['removed_in'] = version
            del values['version']

        return values
```

Reply:
I've changed it back to the toplevel version.